src/main/java/appeng/client/gui/AEBaseGui.java

```diff
@@ -236,21 +257,22 @@ protected final void drawGuiContainerBackgroundLayer( final float f, final int x
 		final List<Slot> slots = this.getInventorySlots();
 		for( final Slot slot : slots )
 		{
-			if( slot instanceof OptionalSlotFake )
+			if( slot instanceof IOptionalSlot )
 			{
-				final OptionalSlotFake fs = (OptionalSlotFake) slot;
-				if( fs.renderDisabled() )
+				final IOptionalSlot optionalSlot = (IOptionalSlot) slot;
+				if( optionalSlot.isRenderDisabled() )
 				{
-					if( fs.isSlotEnabled() )
+					final AppEngSlot aeSlot = (AppEngSlot) slot;
+					if( aeSlot.isSlotEnabled() )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)